rioxarray/rioxarray.py

```diff
@@ -162,7 +178,7 @@ def _get_nonspatial_coords(
                 src_data_array[coord].values,
                 src_data_array[coord].attrs,
             )
-    return coords
+    return xarray.Coordinates(coords)
 
 
 def _make_coords(
```

Comment:
It'd be nice to have an accessor function to set this index even if I opened a Zarr file for example. Is there something like `ds.rio.assign_coords`? I didn't see anything similar in the docs. Perhaps we should have `ds.rio.assign_indexes()` that does both RasterIndex and CRSIndex

Reply:
Do you think that would be a good fit to add to a rasterix accessor? If you do that, you just set `decode_coords=False` and assign the index/coords post load.

Reply:
It needs a bunch of extra info : x dim name, y dim name, transform, and so it's pretty convenient to just do all that with the `.rio` accessor. It seems like a prudent and convenient addition

Reply:
Inside rasterix, it could look for the `.rio` , `.odc`, or `.geo` accessors inside the method. It could be done similar to `open_dataset` where it automatically figures out how to open files based on what's available. It also allows the user select their preferred engine. If they don't exist, then raise an error.

Reply:
That is an option but to me it feels a lot cleaner to add a method under the `rio` namespace. (and similarly under the other namespaces if they thought it was a good idea). At least for rioxarray it's just a small wrapper around an existing function, so it doesn't seem like meaningful extra maintenance burden to me.

Reply:
> Inside rasterix, it could look for the .rio , .odc, or .geo accessors inside the method

Hmm this wouldn't be ideal IMHO, assuming that the goal of the rasterix (and xproj) project is to provide reusable utilities to downstream packages like rioxarray, odc-geo, etc. even though technically looking for those accessors is possible without having cyclic dependencies.

Under the same assumption, rasterix and xproj should also be flexible / not strongly opinionated about the names of the spatial coordinates and dimensions, etc.